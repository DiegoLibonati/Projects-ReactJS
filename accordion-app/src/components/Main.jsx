import React, { useState } from "react";
import questions from "../helpers/data";
import { Question } from "./Question";

export const Main = () => {
  const [question, setQuestion] = useState(questions);

  return (
    <>
      <main className="main_container">
        <section className="faqs_container">
          <article className="faqs_container_title">
            <h1>Questions And Answers About Login</h1>
          </article>

          <article className="faqs_container_questions">
            {question.map((question) => {
              return <Question key={question.id} {...question}></Question>;
            })}
          </article>
        </section>
      </main>
    </>
  );
};
