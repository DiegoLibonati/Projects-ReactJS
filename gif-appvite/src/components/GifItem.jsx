import PropTypes from "prop-types";

export const GifItem = ({
  title,
  url,
  avatar,
  avatarName,
  avatarDescription,
  avatarProfileUrl,
  gifDownload,
  setShowImg,
}) => {
  const handleShowImg = (e) => {
    const imgSrc = e.target.parentNode.parentNode.children[0].children[0].src;
    const imgAlt = e.target.parentNode.parentNode.children[0].children[0].alt;
    setShowImg([imgSrc, imgAlt]);

    const containerShowImg =
      e.target.parentNode.parentNode.parentNode.parentNode.parentNode.parentNode
        .lastChild;

    containerShowImg.style.display = "Flex";
    document.body.style.overflow = "Hidden";
  };

  return (
    <div className="gifs_container_category_list_card">
      <div className="gifs_container_category_list_card_img">
        <img data-testid="test-title" src={url} alt={title}></img>
        <div className="gifs_container_category_list_card_img_description">
          <div className="gifs_container_category_list_card_img_description_header">
            <img src={avatar} alt={avatarName}></img>
            <h3>{avatarName}</h3>
          </div>
          <p>{avatarDescription}</p>
          <a href={avatarProfileUrl} target="_blank">
            Go to user Profile
          </a>
          <a href={gifDownload} target="_blank">
            Download
          </a>
        </div>
      </div>

      <div className="gifs_container_category_list_card_div">
        <p onClick={(e) => handleShowImg(e)}>{title}</p>
      </div>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  avatarName: PropTypes.string.isRequired,
  avatarDescription: PropTypes.string.isRequired,
  avatarProfileUrl: PropTypes.string.isRequired,
  gifDownload: PropTypes.string.isRequired,
  setShowImg: PropTypes.func.isRequired,
};

/* GifItem.defaultProps = {
  // Si no se le pasa nada a las props, se cargan esto de forma default
  title: "Gif",
  url: "...",
}; */
