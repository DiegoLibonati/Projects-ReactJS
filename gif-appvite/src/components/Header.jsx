import React from "react";
import { BsFacebook, BsInstagram, BsGithub } from "react-icons/bs";

export const Header = () => {
  return (
    <>
      <header className="header_container">
        <img
          src="https://raw.githubusercontent.com/DiegoLibonati/Projects-JS/main/Games/breakout/templates/DL.png"
          alt="Logo Diego Libonati"
        ></img>

        <nav className="header_container_nav">
          <ul className="header_container_nav_ul">
            <li className="header_container_nav_ul_li">
              <a href="#">
                <BsFacebook className="icon" />
              </a>
            </li>
            <li className="header_container_nav_ul_li">
              <a href="#">
                <BsInstagram className="icon" />
              </a>
            </li>
            <li className="header_container_nav_ul_li">
              <a href="#">
                <BsGithub className="icon" />
              </a>
            </li>
          </ul>
        </nav>
      </header>
    </>
  );
};
