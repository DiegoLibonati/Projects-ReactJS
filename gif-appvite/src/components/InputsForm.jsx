import React from "react";

export const InputsForm = ({ type, placeholder, value, eventfunc, classN }) => {
  return (
    <input
      className={classN}
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={eventfunc}
    ></input>
  );
};
