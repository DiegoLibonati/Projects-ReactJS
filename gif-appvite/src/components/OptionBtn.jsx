import React from "react";

export const OptionBtn = ({ func, description }) => {
  return <button onClick={() => func()}>{description}</button>;
};
