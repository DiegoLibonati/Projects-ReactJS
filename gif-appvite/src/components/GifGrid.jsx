import { useFetchGif } from "../hooks/useFetchGif";
import { GifItem } from "./GifItem";
import { BsTrash } from "react-icons/bs";
import { removeCategory } from "../helpers/removeCategory";

export const GifGrid = ({
  category,
  categories,
  resetCategory,
  howManyGifs,
  setShowImg,
}) => {
  /* const [counter, setCounter] = useState(0); // PequeÃ±o contador */

  const { imagenes, loading } = useFetchGif(category, howManyGifs);

  const handleDeleteCategory = (e) => {
    removeCategory(e, categories, resetCategory);
  };

  return (
    <>
      <article className="gifs_container_category" id={category}>
        <div className="gifs_container_category_title">
          <h3>{category}</h3>
          <button onClick={(e) => handleDeleteCategory(e)}>
            <BsTrash id="trash" pointerEvents="none"></BsTrash>
          </button>
        </div>
        {/*         <h2>{counter}</h2>
        <button onClick={()=>setCounter(counter+1)}>+1</button> */}

        <ol className="gifs_container_category_list">
          {/**/}
          {loading ? (
            <div className="spinner"></div>
          ) : (
            imagenes.map((imagen) => (
              <GifItem
                key={imagen.id}
                {...imagen}
                setShowImg={setShowImg}
              ></GifItem>
            ))
          )}
        </ol>
      </article>
    </>
  );
};
