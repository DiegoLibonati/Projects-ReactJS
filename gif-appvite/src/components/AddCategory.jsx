import React, { useState } from "react";
import { InputsForm } from "./InputsForm";

export const AddCategory = ({ addCategory, howManyGif, howManyGifs }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleInputNumberChange = (e) => {
    howManyGif(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.length > 0) {
      howManyGif(howManyGifs);
      addCategory(inputValue.trim()); // Trim borra el espacio en blanco adelante y atras
      setInputValue("");
    }
    return;
  };

  return (
    <article className="gifs_container_form">
      <form onSubmit={handleSubmit}>
        <InputsForm
          type="text"
          placeholder="Buscar Gif"
          value={inputValue}
          eventfunc={handleInputChange}
          classN="gifs_container_form_inputValue"
        ></InputsForm>

        <InputsForm
          type="number"
          placeholder=""
          value={howManyGifs}
          eventfunc={handleInputNumberChange}
          classN="gifs_container_form_inputValueNumber"
        ></InputsForm>

        <button type="submit" className="gifs_container_form_button">
          GO
        </button>
      </form>
    </article>
  );
};
