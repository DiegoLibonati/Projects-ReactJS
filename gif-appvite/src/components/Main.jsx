import { useState } from "react";
import { getRandomGifsCategory } from "../helpers/getRandomGifsCategory";
import { removeAllCategories } from "../helpers/removeAllCategories";
import { AddCategory, GifGrid } from "./index";
import { OptionBtn } from "./OptionBtn";

export const Main = () => {
  const [categories, setCategories] = useState([]);
  const [howManyGif, setHowManyGif] = useState(10);
  const [showImg, setShowImg] = useState([]);

  const handleAddCategory = (value) => {
    if (!categories.includes(value)) {
      setCategories([value, ...categories]); // Agregar un dato al array del use state
    }
  };

  const handleRemoveAllCategories = () => {
    removeAllCategories(categories, setCategories);
  };

  const handleSurprise = async () => {
    const getTitle = await getRandomGifsCategory();

    const newTitle = getTitle.split("by");

    if (newTitle == "" || newTitle == " ") {
      handleSurprise();
    } else {
      setHowManyGif(howManyGif);
      handleAddCategory(newTitle[0].trim());
    }
  };

  const showImage = (e) => {
    const showImgContainer = e.target.parentNode.parentNode;

    setShowImg([]);

    showImgContainer.style.display = "None";
    document.body.style.overflow = "auto";
  };
  return (
    <>
      <main>
        <section className="gifs_container">
          {/*Titulo*/}
          <h1>Die GifApp</h1>

          {/*Input*/}
          <AddCategory
            addCategory={handleAddCategory}
            howManyGif={setHowManyGif}
            howManyGifs={howManyGif}
          ></AddCategory>
          {/* Listado de Gif */}

          <article className="gifs_container_options">
            <OptionBtn
              func={handleRemoveAllCategories}
              description="REMOVE ALL CATEGORIES"
            ></OptionBtn>

            <OptionBtn func={handleSurprise} description="SURPRISE"></OptionBtn>
          </article>

          {categories.map((category) => (
            <GifGrid
              key={category}
              category={category}
              categories={categories}
              resetCategory={setCategories}
              howManyGifs={howManyGif}
              setShowImg={setShowImg}
            />
          ))}
        </section>

        <section className="showGif_container">
          <article className="showGif_container_center">
            <img src={showImg[0]} alt={showImg[1]}></img>
            <button onClick={(e) => showImage(e)}>X</button>
          </article>
        </section>
      </main>

      <p data-testid="testcategories">{categories.length}</p>
    </>
  );
};
