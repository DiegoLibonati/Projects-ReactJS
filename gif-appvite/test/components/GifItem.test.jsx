import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("test en <GrifItem>", () => {
  const title = "Dragon Ball";
  const url = "https://dragon-ball.com/goku.jpg";
  /*   test("Debe coincidir con el snapshot", async () => {
    // snapshot del componente

    const container = render(<GifItem title={title} url={url}></GifItem>);

    expect(container).toMatchSnapshot();
  });
 */
  test("Debe encontar un texto", () => {
    const { getByText } = render(<GifItem title={title} url={url} />);

    expect(getByText(title)).toBeTruthy();
  });

  test("Debe encontrar por test-id", () => {
    const { getByTestId } = render(<GifItem title={title} url={url} />);

    expect(getByTestId("test-title").src).toContain(url);
  });

  test("Debe mostrar URL y ALT", () => {
    render(<GifItem title={title} url={url}></GifItem>);

    screen.debug();

    expect(screen.getAllByRole(img).src).toContain(url);
  });
});
