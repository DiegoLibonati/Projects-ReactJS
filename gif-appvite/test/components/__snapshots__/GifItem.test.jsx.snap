// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en <GrifItem> Debe coincidir con el snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="gifs_container_category_list_card"
      >
        <div
          class="gifs_container_category_list_card_img"
        >
          <img
            alt="Dragon Ball"
            src="https://dragon-ball.com/goku.jpg"
          />
          <div
            class="gifs_container_category_list_card_img_description"
          >
            <div
              class="gifs_container_category_list_card_img_description_header"
            >
              <img />
              <h3 />
            </div>
            <p />
            <a
              target="_blank"
            >
              Go to user Profile
            </a>
            <a
              target="_blank"
            >
              Download
            </a>
          </div>
        </div>
        <div
          class="gifs_container_category_list_card_div"
        >
          <p>
            Dragon Ball
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gifs_container_category_list_card"
    >
      <div
        class="gifs_container_category_list_card_img"
      >
        <img
          alt="Dragon Ball"
          src="https://dragon-ball.com/goku.jpg"
        />
        <div
          class="gifs_container_category_list_card_img_description"
        >
          <div
            class="gifs_container_category_list_card_img_description_header"
          >
            <img />
            <h3 />
          </div>
          <p />
          <a
            target="_blank"
          >
            Go to user Profile
          </a>
          <a
            target="_blank"
          >
            Download
          </a>
        </div>
      </div>
      <div
        class="gifs_container_category_list_card_div"
      >
        <p>
          Dragon Ball
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
